plugin-api/src/main/java/org/sonar/api/server/rule/RulesDefinition.java

```diff
@@ -593,8 +619,21 @@ abstract class Rule {
      */
     public abstract RuleType type();
```

Comment:
Shouldn't be `type()` and `severity()` methods be deprecated as well?

Reply:
Intitially, we were thinking to deprecate only the setter method and not the getter, but after discussion it indeed makes more sense to be consistent in deprecating all methods

Reply:
done